### DISCUSSION TITLE
[Feature request] Multiple wiring and QoL feature requests

### DISCUSSION THREAD

**Discussion by HyperB1** [FakeFishGames/Barotrauma#12663] [Ideas and suggestions]
### Here are my suggestions for the little things to add in the new update. They aren't absolutely game-changing, but would genuinely make a great and welcome change for so many people:

1. Set_Expression connector for the RegEx find component.

2. Set_False_Output connector for the Signal check component.

3. Max_Power_Out connector for the reactor (because trying to automatically calculate the reactor's Max power output number for your component-based reactor controller when it changes (with upgrades or talents) by using the wiring components can make it very unprecise sometimes).

4. Connector that outputs the amount of objects detected in the range of the motion detector (IDK what it should be called).

5. Option to change the max string length of the Terminal in the sub editor. (Currently max string length of the Terminal sits at 200 characters by default and cannot be changed)

6. "Oxygen output multiplier" parameter for the vents. Right now all vents connected to one O2 generator output the same proportion of oxygen. Adjusting this parameter would allow the vent to output more or less oxygen relative to others. (When I tried to implement the feature myself I encountered a ton of (mostly mathematical; total oxygen generation didn't add up properly) code problems with this one, so this feature may be difficult to code in).

7. Customizable limits on the number of players on a certain profession (Right now these limits are as follows: Captain - 1 player max, Security & Medic - 2 players max, Engineer, Mechanic, Assistant - No limit). Introducing the option to adjust these limits on servers when needed would be very nice.

**Comment by SomeRandomNoobKekeke** [comment]
3 Wasn't circuit control nerfed? I thought circuit can't outperform standart autopilot

6 I thought vents distribute Oxygen proportionate to the volume of Hull the vent is in, also you can  put more vents in same room if you're suffocating

**Reply by HyperB1** [reply]
- No, circuit still can outperform native auto control, and yes, people still do them.

- Yes, and the multiplier will allow you to change that number

**Comment by pavel4win** [comment]
For Motion Detector
Connector: Angle direction (0-360 degrees)
Connector: distance to nearest object
in the "motion detector" field would make it possible to control part of the detector's circumference. And make automatic guns.
